text/204-golang-bindings.md

```diff
@@ -0,0 +1,669 @@
+
+```ts
+export interface BaseServiceOptions {
+  readonly cluster: ICluster;
+  readonly desiredCount?: number;
+  readonly serviceName?: string;
+  readonly maxHealthyPercent?: number;
+  readonly minHealthyPercent?: number;
+  readonly healthCheckGracePeriod?: Duration;
+  readonly cloudMapOptions?: CloudMapOptions;
+  readonly propagateTags?: PropagatedTagSource;
+  readonly enableECSManagedTags?: boolean;
+  readonly deploymentController?: DeploymentController;
+}
+
+export interface BaseServiceProps extends BaseServiceOptions {
+  readonly launchType: LaunchType;
+}
+```
+
+As before, the Go translation would have the getter methods (e.g. readonly cluster) defined in a the generated interface, which correspond to
+non-exported struct members in the struct that implements each getter.
+
+```go
+package ecs
+
+type BaseServiceOptionsIface interface {
+    Cluster()                ICluster
+    DesiredCount()           *int
+    ServiceName()            *string
+    MaxHealthyPercent()      *int
+    MinHealthyPercent()      *int
+    HealthCheckGracePeriod() *Duration
+    CloudMapOptions()        CloudMapOptions
+    PropagateTags()          PropagatedTagSource
+    EnableECSManagedTags()   *bool
+    DeploymentController()   DeploymentController
+}
+
+type BaseServicePropsIface interface {
+    serviceOptions           BaseServiceOptionsIface  // embeddeded interface
+    LaunchType()             LaunchType
+}
+
+type BaseServiceOptions struct {
+    cluster                ICluster
+    desiredCount           *int
+    serviceName            *string
+    maxHealthyPercent      *int
+    minHealthyPercent      *int
+    healthCheckGracePeriod *Duration
+    cloudMapOptions        CloudMapOptions
+    propagateTags          PropagatedTagSource
+    enableECSManagedTags   *bool
+    deploymentController   DeploymentController
+}
+
+type BaseServiceProps struct {
+    baseOptions         BaseServiceOptions  // embedded (anonymous?) field
+    launchType          LaunchType
+}
+
+func (o BaseServiceOptions) Cluster() ICluster { return o.cluster }
+func (o BaseServiceOptions) DesiredCount() int { return o.desiredCount }
+
+// ... etc
+
+```
+
+This allows the embedding interface (here, `BaseServicePropsIFace`) to "inherit" all the methods defined in the embedded interface
+(`BaseServiceOptionsIFace`). Similarly, the embedding struct (`BaseServiceProps`) would have access to all the properties of the embedded struct
+(`BaseServiceOptions`) through having it as an anonymous field (See: [Go playground example](https://play.golang.org/p/XRpTH3LN6bx)).
+
+```go
+props := BaseServiceProps{
+    BaseServiceOptions{
+      ServiceName: "myService",
+      ...
+    },
+    LaunchType: "EC2",
+}
+
+props.ServiceName  // => "myService"
+```
+
+### Notes/Concerns
+
+* When fields in the Go struct are exported, the struct cannot have a method with the same name (e.g. exported field `Bar` , and method `Bar()`.
```

Comment:
This is a very informative section, but its guidance conflicts with the examples presented earlier. Maybe it would be better to have this notes/concerns sections before the code examples above so that the example style is consistent.

Alternatively would be to write examples above in the proposed style. Alternative design sections, (e.g. appendixes)  could call other potential design considerations.

Reply:
Fixed here https://github.com/aws/aws-cdk-rfcs/pull/206/commits/366cd5d30797bfb63e26f87f704da030adb7f041